master/internal/task/allocation.go

```diff
@@ -332,7 +332,7 @@ func (a *allocation) SetProxyAddress(_ context.Context, address string) error {
 		return nil
 	}
 	a.model.ProxyAddress = &address
-	if err := a.db.UpdateAllocationProxyAddress(a.model); err != nil {
+	if err := db.UpdateAllocationProxyAddress(ctx, a.model); err != nil {
```

Comment:
if for some reason the connection dies and the context is canceled, should we really crash or let the requester retry (via DET_RETRY_CONFIG python side)? if the latter, we should catch `context.Canceled` with `errors.Is` and return `ctx.Err()` but not crash (maybe `context.DeadlineExceeded` so the system under load can try to recover without experiments or notebooks losing their allocations and by that their place in the queue).

Reply:
Ooh good catch of the corner case, I think we should do the latter -- catch `context.Canceled` & not crash. I'll leave this open if someone else has other thoughts.

Reply:
on second thought, let's just return `err` and not crash in all cases - we want the python client to be able to retry on any failure, and we'll crash if it finally gives up and the containers exit.

Reply:
I think you added in the crash in https://github.com/determined-ai/determined/commit/fa626b0ef65aad821ed86551c0b146ba7066e789, so since it's your work, happy to change it back as you like